src/components/ComponentPreview.tsx

```diff
@@ -16,18 +22,38 @@ const JSONPreview: React.FC<JSONPreviewProps> = ({data, className = ''}) => (
   </pre>
 );
 
-export interface ComponentPreviewProps {
-  component: ExtendedEditFormComponentSchema;
-}
+/*
+  Fallback component preview component, for when no suitable preview is defined.
+
+  TODO: I think the types currently require this to be defined, so should this even
+  exist? Maybe we should just throw an exception if you're trying to render a component
+  type that's not in the registry?
+ */
 
 const Fallback: React.FC<ComponentPreviewProps> = ({component}) => <JSONPreview data={component} />;
 
-type PreviewState = 'rich' | 'JSON';
+/*
```

Comment:
yeah I'll take a look! the primary goal was to add structure to the code itself since there are some logical groups identifiable, but the docs tab itself can benefit from this.